app/docker/views/containers/edit/containerController.js

```diff
@@ -192,40 +192,113 @@ function ($q, $scope, $state, $transition$, $filter, Commit, ContainerHelper, Co
     var container = $scope.container;
     var config = ContainerHelper.configFromContainer(container.Model);
     $scope.state.recreateContainerInProgress = true;
-    ContainerService.remove(container, true)
-    .then(function success() {
+    var isRunning = container.State.Running;
+
+    return stopContainerIfNeeded()
+      .then(renameContainer)
+      .then(pullImageIfNeeded)
+      .then(createContainer)
+      .then(connectContainerToOtherNetworks)
+      .then(startContainerIfNeeded)
+      .then(createResourceControlIfNeeded)
+      .then(deleteOldContainer)
+      .then(notifyAndChangeView)
+      .catch(notifyOnError);
+
+    function stopContainerIfNeeded() {
+      if (!isRunning) {
+        return $q.when();
+      }
+      return ContainerService.stopContainer(container.Id);
+    }
+
+    function renameContainer() {
+      return ContainerService.renameContainer(container.Id, container.Name + '-old');
+    }
+
+    function pullImageIfNeeded() {
+      if (!pullImage) {
+        return $q.when();
+      }
+      return getRegistry().then(function pullImage(containerRegistery) {
+        return ImageService.pullImage(container.Config.Image, containerRegistery, true);
+      });
+    }
+
+    function getRegistry() {
+      // get container registery
       return RegistryService.retrieveRegistryFromRepository(container.Config.Image);
-    })
-    .then(function success(data) {
-      return $q.when(!pullImage || ImageService.pullImage(container.Config.Image, data, true));
-    })
-    .then(function success() {
-      return ContainerService.createAndStartContainer(config);
-    })
-    .then(function success(data) {
-      if (!container.ResourceControl) {
-        return true;
-      } else {
-        var containerIdentifier = data.Id;
-        var resourceControl = container.ResourceControl;
-        var users = resourceControl.UserAccesses.map(function(u) {
-          return u.UserId;
-        });
-        var teams = resourceControl.TeamAccesses.map(function(t) {
-          return t.TeamId;
+    }
+
+    function createContainer() {
```

Comment:
You should rename this function to `setMainNetworkAndCreateContainer` and remove the comment below.